src/Ripple/BinaryCoding.h

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+#include <vector>
+
+namespace TW {
+namespace Ripple {
+
+/// Encodes a 16-bit value into the provided buffer.
+static inline void encode16(uint16_t val, std::vector<uint8_t>& data) {
+    data.push_back (static_cast<unsigned char> (val >> 8));
+    data.push_back (static_cast<unsigned char> (val & 0xff));
+}
+
+/// Encodes a 32-bit value into the provided buffer.
+static inline void encode32(uint32_t val, std::vector<uint8_t>& data) {
+    data.push_back(static_cast<uint8_t>((val >> 24)));
+    data.push_back(static_cast<uint8_t>((val >> 16) & 0xff));
+    data.push_back(static_cast<uint8_t>((val >> 8) & 0xff));
+    data.push_back(static_cast<uint8_t>(val));
+}
+
+/// Encodes a 64-bit value into the provided buffer.
+static inline void encode64(uint64_t val, std::vector<uint8_t>& data) {
```

Comment:
We should consolidate this and the one under Bitcoin.

Reply:
Yes, the only difference is endian?

Reply:
Yes, let's have something like `encode64le` and `encode64be`